api/processors/actions/group/discard.py

```diff
@@ -0,0 +1,61 @@
+from django.db import transaction
+
+from api.models import Group
+from api.models.auxil.status_setter import StatusSetter
+from api.models.choices.log_event_type import StudentLogEventType, TeacherLogEventType
+from api.models.choices.status import StudentProjectStatus, TeacherProjectStatus
+from api.processors.actions.group import GroupActionProcessor
+from api.processors.auxil.log_event_creator import GroupLogEventCreator
+
+
+class GroupDiscardProcessor(GroupActionProcessor):
+    def __init__(self, group: Group, reason: str):
+        self.reason = reason
+        super().__init__(group)
+
+    @transaction.atomic
+    def process(self) -> None:
+        self._set_statuses()
+        self._create_log_events()
+        self._delete()
+
+    def _create_log_events(self) -> None:
+        # TODO add from_groupd once CASCAD policy removed
```

Comment:
@Auch-Auch `CASCADE` was replaced during merging of #209, so you can create a small separate PR to create log events.